libdnf-cli/argument_parser.cpp

```diff
@@ -314,13 +338,29 @@ static void print_complete(
 
     // Search for matching commands.
     if (arg[0] == '\0' || arg[0] != '-') {
-        for (const auto * opt : cmd->get_commands()) {
+        for (const auto * opt : get_commands()) {
             auto & name = opt->get_id();
             if (name.compare(0, strlen(arg), arg) == 0) {
                 help.add_line(name, '(' + opt->get_short_description() + ')', nullptr);
                 last = name + ' ';
             }
         }
+        if (last.empty() && used_positional_arguments < get_positional_args().size()) {
+            auto pos_arg = get_positional_args()[used_positional_arguments];
+            if (auto & complete_hook = pos_arg->complete_hook) {
```

Comment:
Why assign to the temporary variable? On a personal note I don't like this syntax and find it confusing, but in this case it doesn't even seem to save anything. I'm pretty sure the access to the member variable pointed to by `pos_arg` will be optimized by compiler, so you can just do:
```C++
if (pos_arg->complete_hook) {
    auto result = pos_arg->complete_hook(arg);
    // ...
```

Reply:
In this case, it is only a reference. And it's a matter of personal taste. If the referenced pointer is non-nullptr, I can use the reference.
But I made a change. It's unimportant to me.